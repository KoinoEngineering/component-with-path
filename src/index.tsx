export const createRoot = (rootPath: string) => { };