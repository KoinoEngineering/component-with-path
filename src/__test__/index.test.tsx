import { render, screen } from "@testing-library/react";
import React from "react";
import { createRoot, WithPathProps } from "src/index";

const Root1 = createRoot("root1");
const Root2 = createRoot("root2");

const EchoPath: React.FC<WithPathProps> = ({ id, usePath, children }) => {
    return <div id={id}>
        <div>{"My Path is : " + usePath()}</div>
        <div>
            {children}
        </div>
    </div>;
};

const EchoPathAndId: React.FC<WithPathProps> = ({ id, usePath, children }) => {
    return <div id={id}>
        <div>{"My Path is : " + usePath() + " / My id is : " + id}</div>
        <div>
            {children}
        </div>
    </div>;
};

const EchoPathWithRoot1 = Root1.withPath(EchoPath);
const EchoPathAndIdWithRoot1 = Root1.withPath(EchoPathAndId);
const EchoPathWithRoot2 = Root2.withPath(EchoPath);
const EchoPathAndIdWithRoot2 = Root2.withPath(EchoPathAndId);

describe("snap shot testings", () => {
    test("renders one root,one component", () => {
        const target = render(<div>
            <EchoPathWithRoot1 id="GrandParent">
                <EchoPathWithRoot1 id="Parent">
                    <EchoPathWithRoot1 id="Me">
                        <EchoPathWithRoot1 id="Child">
                            <EchoPathWithRoot1 id="GrandChild" />
                        </EchoPathWithRoot1>
                    </EchoPathWithRoot1>
                </EchoPathWithRoot1>
            </EchoPathWithRoot1>
        </div>);
        expect(target).toMatchSnapshot();
    });
    test("renders one root,two component", () => {
        const target = render(<div>
            <EchoPathWithRoot1 id="GrandParent">
                <EchoPathAndIdWithRoot1 id="Parent">
                    <EchoPathWithRoot1 id="Me">
                        <EchoPathAndIdWithRoot1 id="Child">
                            <EchoPathWithRoot1 id="GrandChild" />
                        </EchoPathAndIdWithRoot1>
                    </EchoPathWithRoot1>
                </EchoPathAndIdWithRoot1>
            </EchoPathWithRoot1>
        </div>);
        expect(target).toMatchSnapshot();
    });
    test("renders two root", () => {
        const target = render(<div>
            <EchoPathWithRoot1 id="GrandParent">
                <EchoPathAndIdWithRoot1 id="Parent">
                    <EchoPathWithRoot1 id="Me">
                        <EchoPathAndIdWithRoot1 id="Child">
                            <EchoPathWithRoot1 id="GrandChild" />
                        </EchoPathAndIdWithRoot1>
                    </EchoPathWithRoot1>
                </EchoPathAndIdWithRoot1>
            </EchoPathWithRoot1>
            <EchoPathWithRoot2 id="GrandParent">
                <EchoPathAndIdWithRoot2 id="Parent">
                    <EchoPathWithRoot2 id="Me">
                        <EchoPathAndIdWithRoot2 id="Child">
                            <EchoPathWithRoot2 id="GrandChild" />
                        </EchoPathAndIdWithRoot2>
                    </EchoPathWithRoot2>
                </EchoPathAndIdWithRoot2>
            </EchoPathWithRoot2>
        </div>);
    });
    test("renders mixed", () => {
        const target = render(<div>
            <EchoPathWithRoot1 id="GrandParent">
                <EchoPathAndIdWithRoot1 id="Parent">
                    <EchoPathWithRoot2 id="GrandParent">
                        <EchoPathWithRoot1 id="Me">
                            <EchoPathAndIdWithRoot2 id="Parent">
                                <EchoPathWithRoot2 id="Me">
                                    <EchoPathAndIdWithRoot1 id="Child">
                                        <EchoPathWithRoot1 id="GrandChild" />
                                        <EchoPathAndIdWithRoot2 id="Child">
                                            <EchoPathWithRoot2 id="GrandChild" />
                                        </EchoPathAndIdWithRoot2>
                                    </EchoPathAndIdWithRoot1>
                                </EchoPathWithRoot2>
                            </EchoPathAndIdWithRoot2>
                        </EchoPathWithRoot1>
                    </EchoPathWithRoot2>
                </EchoPathAndIdWithRoot1>
            </EchoPathWithRoot1>
        </div>);
        expect(target).toMatchSnapshot();
    });
})
