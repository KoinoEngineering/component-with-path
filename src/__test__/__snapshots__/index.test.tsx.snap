// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snap shot testings renders mixed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          id="GrandParent"
        >
          <div>
            My Path is : root1.GrandParent
          </div>
          <div>
            <div
              id="Parent"
            >
              <div>
                My Path is : root1.GrandParent.Parent / My id is : Parent
              </div>
              <div>
                <div
                  id="GrandParent"
                >
                  <div>
                    My Path is : root2.GrandParent
                  </div>
                  <div>
                    <div
                      id="Me"
                    >
                      <div>
                        My Path is : root1.GrandParent.Parent.Me
                      </div>
                      <div>
                        <div
                          id="Parent"
                        >
                          <div>
                            My Path is : root2.GrandParent.Parent / My id is : Parent
                          </div>
                          <div>
                            <div
                              id="Me"
                            >
                              <div>
                                My Path is : root2.GrandParent.Parent.Me
                              </div>
                              <div>
                                <div
                                  id="Child"
                                >
                                  <div>
                                    My Path is : root1.GrandParent.Parent.Me.Child / My id is : Child
                                  </div>
                                  <div>
                                    <div
                                      id="GrandChild"
                                    >
                                      <div>
                                        My Path is : root1.GrandParent.Parent.Me.Child.GrandChild
                                      </div>
                                      <div />
                                    </div>
                                    <div
                                      id="Child"
                                    >
                                      <div>
                                        My Path is : root2.GrandParent.Parent.Me.Child / My id is : Child
                                      </div>
                                      <div>
                                        <div
                                          id="GrandChild"
                                        >
                                          <div>
                                            My Path is : root2.GrandParent.Parent.Me.Child.GrandChild
                                          </div>
                                          <div />
                                        </div>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        id="GrandParent"
      >
        <div>
          My Path is : root1.GrandParent
        </div>
        <div>
          <div
            id="Parent"
          >
            <div>
              My Path is : root1.GrandParent.Parent / My id is : Parent
            </div>
            <div>
              <div
                id="GrandParent"
              >
                <div>
                  My Path is : root2.GrandParent
                </div>
                <div>
                  <div
                    id="Me"
                  >
                    <div>
                      My Path is : root1.GrandParent.Parent.Me
                    </div>
                    <div>
                      <div
                        id="Parent"
                      >
                        <div>
                          My Path is : root2.GrandParent.Parent / My id is : Parent
                        </div>
                        <div>
                          <div
                            id="Me"
                          >
                            <div>
                              My Path is : root2.GrandParent.Parent.Me
                            </div>
                            <div>
                              <div
                                id="Child"
                              >
                                <div>
                                  My Path is : root1.GrandParent.Parent.Me.Child / My id is : Child
                                </div>
                                <div>
                                  <div
                                    id="GrandChild"
                                  >
                                    <div>
                                      My Path is : root1.GrandParent.Parent.Me.Child.GrandChild
                                    </div>
                                    <div />
                                  </div>
                                  <div
                                    id="Child"
                                  >
                                    <div>
                                      My Path is : root2.GrandParent.Parent.Me.Child / My id is : Child
                                    </div>
                                    <div>
                                      <div
                                        id="GrandChild"
                                      >
                                        <div>
                                          My Path is : root2.GrandParent.Parent.Me.Child.GrandChild
                                        </div>
                                        <div />
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`snap shot testings renders one root,one component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          id="GrandParent"
        >
          <div>
            My Path is : root1.GrandParent
          </div>
          <div>
            <div
              id="Parent"
            >
              <div>
                My Path is : root1.GrandParent.Parent
              </div>
              <div>
                <div
                  id="Me"
                >
                  <div>
                    My Path is : root1.GrandParent.Parent.Me
                  </div>
                  <div>
                    <div
                      id="Child"
                    >
                      <div>
                        My Path is : root1.GrandParent.Parent.Me.Child
                      </div>
                      <div>
                        <div
                          id="GrandChild"
                        >
                          <div>
                            My Path is : root1.GrandParent.Parent.Me.Child.GrandChild
                          </div>
                          <div />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        id="GrandParent"
      >
        <div>
          My Path is : root1.GrandParent
        </div>
        <div>
          <div
            id="Parent"
          >
            <div>
              My Path is : root1.GrandParent.Parent
            </div>
            <div>
              <div
                id="Me"
              >
                <div>
                  My Path is : root1.GrandParent.Parent.Me
                </div>
                <div>
                  <div
                    id="Child"
                  >
                    <div>
                      My Path is : root1.GrandParent.Parent.Me.Child
                    </div>
                    <div>
                      <div
                        id="GrandChild"
                      >
                        <div>
                          My Path is : root1.GrandParent.Parent.Me.Child.GrandChild
                        </div>
                        <div />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`snap shot testings renders one root,two component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          id="GrandParent"
        >
          <div>
            My Path is : root1.GrandParent
          </div>
          <div>
            <div
              id="Parent"
            >
              <div>
                My Path is : root1.GrandParent.Parent / My id is : Parent
              </div>
              <div>
                <div
                  id="Me"
                >
                  <div>
                    My Path is : root1.GrandParent.Parent.Me
                  </div>
                  <div>
                    <div
                      id="Child"
                    >
                      <div>
                        My Path is : root1.GrandParent.Parent.Me.Child / My id is : Child
                      </div>
                      <div>
                        <div
                          id="GrandChild"
                        >
                          <div>
                            My Path is : root1.GrandParent.Parent.Me.Child.GrandChild
                          </div>
                          <div />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        id="GrandParent"
      >
        <div>
          My Path is : root1.GrandParent
        </div>
        <div>
          <div
            id="Parent"
          >
            <div>
              My Path is : root1.GrandParent.Parent / My id is : Parent
            </div>
            <div>
              <div
                id="Me"
              >
                <div>
                  My Path is : root1.GrandParent.Parent.Me
                </div>
                <div>
                  <div
                    id="Child"
                  >
                    <div>
                      My Path is : root1.GrandParent.Parent.Me.Child / My id is : Child
                    </div>
                    <div>
                      <div
                        id="GrandChild"
                      >
                        <div>
                          My Path is : root1.GrandParent.Parent.Me.Child.GrandChild
                        </div>
                        <div />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
